workflows/autometa.sh

```diff
@@ -145,20 +176,162 @@ autometa-taxonomy-majority-vote --lca $lca --output $votes --dbdir $ncbi
 
 # output:
 # Will write recovered superkingdoms to ${outdir}
-# e.g. ${outdir}/${simpleName}.bacteria.fna
-# e.g. ${outdir}/${simpleName}.archaea.fna
-# e.g. ${outdir}/${simpleName}.taxonomy.tsv
+# e.g. ${outdir}/${prefix}.bacteria.fna
+# e.g. ${outdir}/${prefix}.archaea.fna
+# e.g. ${outdir}/${prefix}.taxonomy.tsv
 
 # script:
+set -x
 autometa-taxonomy \
     --votes $votes \
     --output $outdir \
-    --prefix $simpleName \
+    --prefix $prefix \
     --split-rank-and-write superkingdom \
     --assembly $assembly \
-    --ncbi $ncbi
+    --dbdir $ncbi \
+    --dbtype $dbtype
+{ set +x; } 2>/dev/null
 
-# Step 5: Perform k-mer counting on respective kingdoms
+# Step 5: Taxon-assignment using the GTDB database 
+# NOTE: only performed if `taxa_routine` is 'ncbi_gtdb'
+
+# Step 5.1: Extract bacterial ORFs and run GTDB
+# input:
+# $kingdom_fasta --> Generated by step 4.3
+# $orfs --> User Input
+
+# output:
+# contig_ids --> text file containing metagenome contig IDs classified within NCBI bacteria and archaea
+# orf_ids --> text file containing contig ORF IDs classified within NCBI bacteria and archaea
+# kingdom_orfs --> fasta file containing metagenome ORFs classified within NCBI bacteria or archaea
+# gtdb_input_orfs --> metagenome orfs classified within NCBI bacteria *and* archaea
+
+if [[ "$taxa_routine" == "ncbi_gtdb" ]]
+then
+    echo "Running GTDB taxon assignment step."
+    # output
+    gtdb_input_orfs="${outdir}/${prefix}.orfs.faa"
+    
+
+    for kingdom in ${kingdoms[@]};do
+
+        kingdom_fasta="${outdir}/${prefix}.${kingdom}.fna"
+        
+        contig_ids="${outdir}/${prefix}.${kingdom}.contigIDs.txt"
+        orf_ids="${outdir}/${prefix}.${kingdom}.orfIDs.txt"
+        kingdom_orfs="${outdir}/${prefix}.${kingdom}.orfs.faa"
+
+        if [ ! -f $kingdom_fasta ]
+        then
+            echo "${kingdom_fasta} does not exist, skipping..."
+            continue
+        fi
+
+        # Retrieve contig IDs from kingdom fasta file
+        set -x
+        grep ">" $kingdom_fasta | \
+            sed 's/^>//' | \
+            sed 's/$/_/' | \
+            cut -f1 -d" " > $contig_ids
+        # Retrieve ORF IDs from contig IDs
+        grep -f $contig_ids $orfs | sed 's/^>//' | cut -f1 -d" " > $orf_ids
```

Comment:
NOTE: See comment above

```suggestion
        grep -f $orf_prefixes $orfs | sed 's/^>//' | cut -f1 -d" " > $orf_ids
```